backend/de.metas.business/src/main/java/de/metas/project/budget/BudgetProjectResource.java

```diff
@@ -22,60 +22,24 @@
 
 package de.metas.project.budget;
 
-import de.metas.calendar.util.CalendarDateRange;
-import de.metas.money.Money;
-import de.metas.product.ResourceId;
 import de.metas.project.ProjectId;
-import de.metas.quantity.Quantity;
-import de.metas.resource.ResourceGroupId;
-import de.metas.uom.UomId;
 import lombok.Builder;
 import lombok.NonNull;
 import lombok.Value;
 
-import javax.annotation.Nullable;
-
 @Value
+@Builder(toBuilder = true)
 public class BudgetProjectResource
 {
-	@NonNull BudgetProjectResourceId id;
-
-	@NonNull ResourceGroupId resourceGroupId;
-	@Nullable ResourceId resourceId;
-
-	@NonNull UomId durationUomId;
-	@NonNull Money plannedAmount;
-	@NonNull Quantity plannedDuration;
-	@NonNull Money pricePerDurationUnit;
+	@NonNull
+	BudgetProjectResourceId id;
 
-	@NonNull CalendarDateRange dateRange;
+	@NonNull
+	BudgetProjectResourceData budgetProjectResourceData;
```

Comment:
IMHO introducing this "Data" object to be used in the actual model and also in upsert Request it's not the best design.

Why?
* because not all the fields of the model are updatable from outside. Maybe they are now, but they won't be in the future. So what will you do then? You will move the not-updatable field from BudgetProjectResourceData directly into BudgetProjectResource and the whole BudgetProjectResource will be a mambo-jambo of fields here and there. Does not feel right.
* the modal is very tightly coupled to the data importing. So changing/improving the data import will influence the code which actually works with those models and vice-versa.
* While it feels super easy to work with imports, it's a nightmare for developers which have to deal with actual data.
* import/export shall not shape the model design

I would suggest:
* have the model as it was before (without the data object)
* have dedicated methods which updates the records from model or from request. Even if it looks like copy-paste, and feels like breaking DRY principle, IMHO this is one of the cases you have too. Btw, here is a nice video explaining a bit more about DRY pros/cons: https://www.youtube.com/watch?v=cqKGDpnE4eY&t=1s

cc @metas-ts